crates/mir/src/lower/function.rs

```diff
@@ -726,7 +729,15 @@ impl<'db, 'a> BodyLowerHelper<'db, 'a> {
     fn lower_expr_to_value(&mut self, expr: &Node<ast::Expr>) -> ValueId {
         let ty = self.expr_ty(expr);
         let inst = self.lower_expr(expr);
-        self.map_to_tmp(inst, ty)
+        let value = self.map_to_tmp(inst, ty);
+
+        // Load primitive types onto stack
+        if ty.is_ptr(self.db) && ty.deref(self.db).is_primitive(self.db) {
+            let inst = self.builder.load(value, expr.into());
+            self.map_to_tmp(inst, ty.deref(self.db))
```

Comment:
I think the `load` insertion should be performed in the `lower_expr` or another transformation pass because `lower_expr_to_value` is not always called and the mapping `value` to `inst` can be performed manually in some AST -> MIR lowering. e.g, `Assign` lowering.

So as a result, the below example is lowered into the inconsistent MIR regarding `SPtr` handling.
```
contract Foo {
    s: Array<i32, 20>

    pub fn test(self) -> i32 {
        let elem: i32 = self.s[0]
        return elem
    }
}
```
↓

![graphviz](https://user-images.githubusercontent.com/6376004/186244580-f2b9f552-4576-4b51-ac03-ffdb0cc9db99.svg)

This inconsistency is not a problem for now because the ugly heauristic in `codegen::yul::isel::function::assign_inst_result` handles this.

But this inconsistency is error-prone, so I think it's better to address this in this PR.
Do you think it's worth adding another transformation pass to perform these heuristics after lowering?